### ISSUE TITLE
Dependency issue with antd v5.0.0

### ISSUE THREAD

**Issue by alenocera** [ant-design/ant-design-charts#1689]
Hello,
just founded this issue while upgrading my project with the new v5.0.0 released today.

`npm ERR! Found: antd@5.0.0
npm ERR! node_modules/antd
npm ERR!   antd@"^5.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer antd@"^4.6.3" from @ant-design/charts@1.4.2
npm ERR! node_modules/@ant-design/charts
npm ERR!   @ant-design/charts@"^1.4.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.`

As suggests by NPM, we can able to continue installation with 
`--legacy-peer-deps`

but I guess a fix is necessary to able to do not have strange issue and strange behaviour at all.

Thanks, appreciate it.

**Comment by lxfu1** [comment]
It will be supported in the near future. If the flowchart is not used, you can use sub packages to pass it first，like: `@ant-design/plots` `@ant-design/graphs`.

**Comment by b5414** [comment]
Same

**Comment by niranjanborawake** [comment]
Hello Team

`@ant-design/charts` v1.4.2 still has the issue in compilation.
```
Failed to compile.

./node_modules/@ant-design/charts/node_modules/@antv/x6-react-components/es/color-picker/index.js
Module not found: Can't resolve 'antd/es/popover/style/index.css' in '<repo>/node_modules/@ant-design/charts/node_modules/@antv/x6-react-components/es/color-picker'

```

**Comment by osabros** [comment]
Same

So far I'm using it with antd5 and all works

The main breaking changes in antd5 is the changing of Moment library to Dayjs, could this impact the charts library?

**Comment by letrape** [comment]
> Hello Team
> 
> `@ant-design/charts` v1.4.2 still has the issue in compilation.
> 
> ```
> Failed to compile.
> 
> ./node_modules/@ant-design/charts/node_modules/@antv/x6-react-components/es/color-picker/index.js
> Module not found: Can't resolve 'antd/es/popover/style/index.css' in '<repo>/node_modules/@ant-design/charts/node_modules/@antv/x6-react-components/es/color-picker'
> ```

I have the same issue, any work around to use antd charts with antd 5, i dont want to downgrade to 4.6

**Comment by Orkuml** [comment]
I'm waiting about solution too. =(

![image](https://user-images.githubusercontent.com/32021798/210799587-0e47880c-e586-4748-a86f-9dc8b2fd8cf3.png)

**Comment by alenocera** [comment]
Guys, as you know the command `--legacy-peer-deps` is not the best solution and if you want to allow compile without using that one, you can use the "overrides" object into your package, like this:

```
"overrides": {
    "@ant-design/charts": {
      "antd": "^5.0.7"
    }
  },
```
Obviously also this solution is not the better one, but avoid you to use the legacy command. Meantime something moves forward to fix this, I'm able to compile a little bit better (and faster) instead using the legacy things.

Hope help you and waiting for final solution.

**Comment by RosaErick** [comment]
Waiting a solution since last year. 😭


![image](https://user-images.githubusercontent.com/49876146/214381474-8340cb73-9546-405a-b5d7-d61cbe8d2018.png)

**Comment by d00rsfan** [comment]
> Guys, as you know the command `--legacy-peer-deps` is not the best solution and if you want to allow compile > ```
> "overrides": {
>     "@ant-design/charts": {
>       "antd": "^5.0.7"
>     }
>   },
I also needed to add this line
      "@ant-design/icons@": "^5.0.0"

**Comment by ildfreelancer** [comment]
For those who got this issue,

While waiting for new upgrade, you can do like this

put this in package.json, you might need to change version to match your current packages
```
  "overrides": {
    "@ant-design/charts": {
      "antd": "^5.3.1",
      "@ant-design/icons": "^5.0.0"
    }
  }
```

**Comment by vgs18** [comment]
Hi, I am also facing the mentioned issue, any leads on when this will be fixed by ant charts team? The workaround might give a temporary fix, but what is the expected date of permanent resolution?
Thanks in advance.

**Comment by arifszn** [comment]
No update still :(

**Comment by hassssan051** [comment]
The library looks great, would love to see an update to v5 so that I can start using it.

**Comment by ranniere-farias-creathus** [comment]
Almost 5 months passed away since my last try, antD is now in the version 5.4.2, nobody fixed this yet! Everyone needs to use antD Charts can't use antD 5.4.2, seriously?

**Comment by Orkuml** [comment]
Almost 5 months passed away since my last try, antD is now in the version 5.4.2, nobody fixed this yet! Everyone needs to use antD Charts can't use antD 5.4.2, seriously?

**Comment by jdlms** [comment]
Like others here I also fixed this issue by removing the charts package and npm installing @ant-design/plots

**Comment by gujiwuqing** [comment]
> 嗨，我也面临着提到的问题，蚂蚁图表团队何时会解决这个问题的任何线索？解决方法可能会提供临时修复，但永久解决的预期日期是什么时候？提前谢谢。

着实有点离谱 我看这个issues已经大半年了 还没解决

**Comment by gabuvi7** [comment]
yes, but it doesn't the way to fix something. I wanna use the line graph and i need the '@ant-design/charts' package. 8 month with the same issue. We are in June and it doesn't seems to be fixed

**Comment by RosaErick** [comment]
August and no update still :(

**Comment by kamivisionvaibhav** [comment]
Hi Team, thank you for all the contribution and making this library free and open source but is there any plan/timeline on next update where this package will be compatible with v5?

**Comment by Ranzeb** [comment]
Please can you fix this issue? It's quite annoying

**Comment by RusseII** [comment]
I would love for this to be fixed thank you for all of your effort !

**Comment by ketangupta34** [comment]
PLEASE FIX THIS, its been a year 🙄

**Comment by AnandD17** [comment]
please fix this issue its been more then a year

**Comment by Tantatorn-dev** [comment]
Any update on this

**Comment by schidobvu** [comment]
Please fix this, More than a year Now!

**Comment by lxfu1** [comment]
If only statistical charts are used, it can be upgraded to V2，We will gradually improve other types of  charts(flowchart、graphs and so on)

**Comment by hustcc** [comment]
> 这个库还维护吗，这个antd的依赖版本问题一年多了都没修。库不维护的话可以标记一下deprecated，以免还有新项目误用这个无人维护的库

这不是已经在升级吗？目前已经把 plot 部分升级了，发布了 2.0 版本，其他的 graph、map 的升级，也需要等待底层升级一起，会逐步发布 2.1、2.2 版本，因为跟随依赖，所以周期会比较长。

如果质疑进度和投入，可以进来一起了解下情况，再看时间是不是太长，是不是没有维护！也期望社区同学能一起进来弄，毕竟是业余之外做开源。

**Comment by hustcc** [comment]
> pps：临时解法，package.json加一下：
> 
> ```
> "overrides": {
>     "@ant-design/charts": {
>       "antd": "^5.11.2",
>       "@ant-design/icons": "^5.2.6"
>     }
>   },
> ```

给解决方案，点赞 👍🏻

**Comment by libinglong** [comment]
这不是开源项目吗？人家100年不维护又怎么样呢？

**Comment by seilkhanovk** [comment]
I'm really concerned about the unresolved issue here. It's becoming evident that many are affected, and with the shift towards antd 5.0, I anticipate even more will encounter this problem. Could you please provide some insight into what's causing the delay in resolving this? Are there unseen challenges that we're not aware of? Any information or updates on your plans to tackle this issue would be greatly appreciated. Thank you for your attention to this matter

**Comment by lxfu1** [comment]
We only have `Flowchart` depending on  `antd`. If you don't use this chart, you can choose to install the sub-packages `@ant-design/[plots|graphs|maps]` to avoid this error. Currently, [XFlow](https://github.com/antvis/XFlow) 2.0 does not have an official website yet, so there should be some stability issues and it cannot be upgraded in a short period of time. Meanwhile, we are also developing Charts 2.0. We have already completed the upgrade of statistical charts and released version [2.0.3](https://ant-design-charts-next.antgroup.com/). The next step is to upgrade the relationship chart. Please follow V2 branch for updates.

**Comment by xsjcTony** [comment]
加油 目前用的antd5 虽然还没用charts 但估计快用到了 💚

**Comment by MatteoToqua** [comment]
Also facing this issue, would appreciate some news from the team.
Thank you!!

**Comment by Tianlin-W** [comment]
2024年的解决办法，@ant-design/charts 升级到 2.x。

**Comment by Tianlin-W** [comment]
2024年的解决办法，@ant-design/charts 升级到 2.x。
